marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.sjs

```diff
@@ -345,6 +347,23 @@ function removeNullProperties(obj) {
   return obj
 }
 
+// Used in UI (currently only in custom step for showing 'Additional Settings' field)
+function getStepWithAdditionalSettings(artifactType, artifactName) {
+  let dataHubOptions = ["sourceQuery", "sourceQueryIsScript", "constrainSourceQueryToJob", "provenanceGranularityLevel", "acceptsBatch",
```

Comment:
It doesn't seem like matchOptions and mergeOptions should really be here, because those are specific to concrete step types. Same with validateEntity (though technically that could be supported for other step types in the future, while it's just mapping today). 

I think this is also worth keeping specific to a custom step for now, and thus shouldn't be in "artifacts/core.sjs" which is starting to have a scope of "anything that a user can persist". I think this library needs to be focused on generic persistence functions as opposed to functions that are specific to a concrete type.